In [168]:
import pandas as pd
import numpy as np
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.graph_objects as go
import statsmodels

In [112]:
df = pd.read_csv("Speed_Dating_Data.csv", encoding= 'iso-8859-1')
pd.options.display.max_columns = None
pd.options.display.max_rows = None

In [113]:
df.head(5)

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

In [114]:
print(df["match"].value_counts())
print(df["dec_o"].value_counts())

match
0    6998
1    1380
Name: count, dtype: int64
dec_o
0    4863
1    3515
Name: count, dtype: int64


In [115]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8378 entries, 0 to 8377
Columns: 195 entries, iid to amb5_3
dtypes: float64(174), int64(13), object(8)
memory usage: 12.5+ MB


In [116]:
df.describe(include="all")

iid           id       gender          idg       condtn  \
count   8378.000000  8377.000000  8378.000000  8378.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     283.675937     8.960248     0.500597    17.327166     1.828837   
std      158.583367     5.491329     0.500029    10.940735     0.376673   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      154.000000     4.000000     0.000000     8.000000     2.000000   
50%      281.000000     8.000000     1.000000    16.000000     2.000000   
75%      407.000000    13.000000     1.000000    26.000000     2.000000   
max      552.000000    22.000000     1.000000    44.000000     2.000000   

               wave        round     position     positin1        order  \
count   8378.000000  8378.000000  8378.000000  6532.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      11.350919    16.872046     9.042731     9.295775     8.927668   
std        5.995903     4.358458     5.514939     5.650199     5.477009   
min        1.000000     5.000000     1.000000     1.000000     1.000000   
25%        7.000000    14.000000     4.000000     4.000000     4.000000   
50%       11.000000    18.000000     8.000000     9.000000     8.000000   
75%       15.000000    20.000000    13.000000    14.000000    13.000000   
max       21.000000    22.000000    22.000000    22.000000    22.000000   

            partner          pid        match     int_corr     samerace  \
count   8378.000000  8368.000000  8378.000000  8220.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       8.963595   283.863767     0.164717     0.196010     0.395799   
std        5.491068   158.584899     0.370947     0.303539     0.489051   
min        1.000000     1.000000     0.000000    -0.830000     0.000000   
25%        4.000000   154.000000     0.000000    -0.020000     0.000000   
50%        8.000000   281.000000     0.000000     0.210000     0.000000   
75%       13.000000   408.000000     0.000000     0.430000     1.000000   
max       22.000000   552.000000     1.000000     0.910000     1.000000   

              age_o       race_o     pf_o_att     pf_o_sin     pf_o_int  \
count   8274.000000  8305.000000  8289.000000  8289.000000  8289.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      26.364999     2.756653    22.495347    17.396867    20.270759   
std        3.563648     1.230689    12.569802     7.044003     6.782895   
min       18.000000     1.000000     0.000000     0.000000     0.000000   
25%       24.000000     2.000000    15.000000    15.000000    17.390000   
50%       26.000000     2.000000    20.000000    18.370000    20.000000   
75%       28.000000     4.000000    25.000000    20.000000    23.810000   
max       55.000000     6.000000   100.000000    60.000000    50.000000   

           pf_o_fun     pf_o_amb     pf_o_sha        dec_o       attr_o  \
count   8280.000000  8271.000000  8249.000000  8378.000000  8166.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      17.459714    10.685375    1

In [117]:
df.head(5)

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

### Apprenons à connaitre la composition de notre dataset avant de se lancer dans l'exploration des données

In [118]:
print(df.isna().sum().to_string())

iid            0
id             1
gender         0
idg            0
condtn         0
wave           0
round          0
position       0
positin1    1846
order          0
partner        0
pid           10
match          0
int_corr     158
samerace       0
age_o        104
race_o        73
pf_o_att      89
pf_o_sin      89
pf_o_int      89
pf_o_fun      98
pf_o_amb     107
pf_o_sha     129
dec_o          0
attr_o       212
sinc_o       287
intel_o      306
fun_o        360
amb_o        722
shar_o      1076
like_o       250
prob_o       318
met_o        385
age           95
field         63
field_cd      82
undergra    3464
mn_sat      5245
tuition     4795
race          63
imprace       79
imprelig      79
from          79
zipcode     1064
income      4099
goal          79
date          97
go_out        79
career        89
career_c     138
sports        79
tvsports      79
exercise      79
dining        79
museums       79
art           79
hiking        79
gaming        79
clubbing      

In [119]:
# Résumé des valeurs du dataset
print("Pour résumer nous avons:")
nombre_de_lignes, nombre_de_colonnes = df.shape
total_na = df.isna().sum().sum()
poids_na = total_na*100/df.size
top_50_na = df.isna().sum().nlargest(50)

print("> Nombre total de valeur :", df.size)
print(f"> Nombre de lignes : {nombre_de_lignes}")
print(f"> Nombre de colonnes : {nombre_de_colonnes}")
print("> Valeurs manquantes", total_na,"soit", poids_na.round(2), "% du dataset")
fig=px.bar(top_50_na, color="value",title="Les 50 premières données ayant le plus de valeur manquante")
fig.show()

Pour résumer nous avons:
> Nombre total de valeur : 1633710
> Nombre de lignes : 8378
> Nombre de colonnes : 195
> Valeurs manquantes 432016 soit 26.44 % du dataset


#### Nous avons énormément de valeur manquante, plus d'1/4 des valeurs du dataset avec 26.44%.
#### Les attributs finissant par "_3" montrent une grandes quantitées de données absentes. Ces colonnes correspondent à un retour de la part des participants 3/4 semaines après l'évènement. On peut donc supposer que cette absence de données et liée à l'absence de retour du questionnaire de la part des candidats au bout de ces 3/4 semaines. 
#### Parmis ces colonnes la "num_in_3" et "numdat_3" montrent le plus faible retour de donnée. Cela s'explique par le fait que ces colonnes sont à renseigner si il y a eu un rendez vous, information attendu sur la colonne "date_3".


In [120]:
print("Répartitions des réponses", df["date_3"].value_counts())
print("Nombre de valeur manquante", df["date_3"].isna().value_counts())

Répartitions des réponses date_3
0.0    2476
1.0    1498
Name: count, dtype: int64
Nombre de valeur manquante date_3
True     4404
False    3974
Name: count, dtype: int64


#### La somme des Non (2476) + la somme des NaN dans la colonne "date_3" est égale à 6880 soit pratiqument la somme des NaN de la colonne "numdate_3" (6882), ce qui est normal car si le candidat repond Non à "date_3" il ne doit pas répondre à cette question et par conséquence aussi à la colonne "num_in_3".

# Commençons à explorer la donnée

#### Nous allons créer un dataset n'ayant que les iid unique afin d'avoir bien des données de participant unique
#### Nous prendrons le temps au fur et à mesure de modifier certaines colonne pour une meilleur lecture des variables

In [121]:
# Modificationde colonne pour une meilleur compréhension des valeurs
df["dec_o"] = df["dec_o"].apply( lambda x : "Non" if x==0 else "Oui" if x==1 else x)
df["gender"] = df["gender"].apply( lambda x : "Woman" if x==0 else "Man" if x==1 else x)
df["samerace"] = df["samerace"].apply(lambda x : "Non" if x==0 else "Oui" if x==1 else x)
df["dec"]=df["dec"].apply(lambda x: "Non" if x==0 else "Oui" if x==1 else x)
df["date_3"]=df["date_3"].apply(lambda x:"Non" if x==0 else "Oui" if x==1 else x)

df.head()

iid   id gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0  Woman    1       1     1     10         7       NaN      4   
1    1  1.0  Woman    1       1     1     10         7       NaN      3   
2    1  1.0  Woman    1       1     1     10         7       NaN     10   
3    1  1.0  Woman    1       1     1     10         7       NaN      5   
4    1  1.0  Woman    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr samerace  age_o  race_o  pf_o_att  pf_o_sin  \
0        1  11.0      0      0.14      Non   27.0     2.0      35.0      20.0   
1        2  12.0      0      0.54      Non   22.0     2.0      60.0       0.0   
2        3  13.0      1      0.16      Oui   22.0     4.0      19.0      18.0   
3        4  14.0      1      0.61      Non   23.0     2.0      30.0       5.0   
4        5  15.0      1      0.21      Non   24.0     3.0      30.0      10.0   

   pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha dec_o  attr_o  sinc_o  intel_o  \
0      20.0      20.0       0.0       5.0   Non     6.0     8.0      8.0   
1       0.0      40.0       0.0       0.0   Non     7.0     8.0     10.0   
2      19.0      18.0      14.0      12.0   Oui    10.0    10.0     10.0   
3      15.0      40.0       5.0       5.0   Oui     7.0     8.0      9.0   
4      20.0      10.0      10.0      20.0   Oui     8.0     7.0      9.0   

   fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd undergra  \
0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0      NaN   
1    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0      NaN   
2   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0      NaN   
3    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0      NaN   
4    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0      NaN   

  mn_sat tuition  race  imprace  imprelig     from zipcode     income  goal  \
0    NaN     NaN   4.0      2.0       4.0  Chicago  60,521  69,487.00   2.0   
1    NaN     NaN   4.0      2.0       4.0  Chicago  60,521  69,487.00   2.0   
2    NaN     NaN   4.0      2.0       4.0  Chicago  60,521  69,487.00   2.0   
3    NaN     NaN   4.0      2.0       4.0  Chicago  60,521  69,487.00   2.0   
4    NaN     NaN   4.0      2.0       4.0  Chicago  60,521  69,487.00   2.0   

   date  go_out  career  career_c  sports  tvsports  exercise  dining  \
0   7.0     1.0  lawyer       NaN     9.0       2.0       8.0     9.0   
1   7.0     1.0  lawyer       NaN     9.0       2.0       8.0     9.0   
2   7.0     1.0  lawyer       NaN     9.0       2.0       8.0     9.0   
3   7.0     1.0  lawyer       NaN     9.0       2.0       8.0     9.0   
4   7.0     1.0  lawyer       NaN     9.0       2.0       8.0     9.0   

   museums  art  hiking  gaming  clubbing  reading   tv  theater  movies  \
0      1.0  1.0     5.0     1.0       5.0      6.0  9.0      1.0    10.0   
1      1.0  1.0     5.0     1.0       5.0      6.0  9.0      1.0    10.0   
2      1.0  1.0     5.0     1.0       5.0      6.0  9.0      1.0    10.0   
3      1.0  1.0     5.0     1.0       5.0      6.0  9.0      1.0    10.0   
4      1.0  1.0     5.0     1.0       5.0      6.0  9.0      1.0    10.0   

   concerts  music  shopping  yoga  exphappy  expnum  attr1_1  sinc1_1  \
0      10.0    9.0       8.0   1.0       3.0     2.0     15.0     20.0   
1      10.0    9.0       8.0   1.0       3.0     2.0     15.0     20.0   
2      10.0    9.0       8.0   1.0       3.0     2.0     15.0     20.0   
3      10.0    9.0       8.0   1.0       3.0     2.0     15.0     20.0   
4      10.0    9.0       8.0   1.0       3.0     2.0     15.0     20.0   

   intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  intel4_1  fun4_1  \
0      20.0    15.0    15.0     15.0      NaN      NaN       NaN     NaN   
1      20.0    15.0    15.0     15.0      NaN      NaN       NaN     NaN   
2      20.0    15.0    15.0     15.0      NaN      NaN       NaN     NaN   
3

In [122]:
df_unique = df.drop_duplicates(subset='iid')


In [123]:
df_unique.head()

iid   id gender  idg  condtn  wave  round  position  positin1  order  \
0     1  1.0  Woman    1       1     1     10         7       NaN      4   
10    2  2.0  Woman    3       1     1     10         3       NaN     10   
20    3  3.0  Woman    5       1     1     10         9       NaN      6   
30    4  4.0  Woman    7       1     1     10         6       NaN      3   
40    5  5.0  Woman    9       1     1     10         4       NaN      1   

    partner   pid  match  int_corr samerace  age_o  race_o  pf_o_att  \
0         1  11.0      0      0.14      Non   27.0     2.0      35.0   
10        1  11.0      0      0.29      Oui   27.0     2.0      35.0   
20        1  11.0      0     -0.24      Oui   27.0     2.0      35.0   
30        1  11.0      0     -0.18      Oui   27.0     2.0      35.0   
40        1  11.0      0     -0.43      Oui   27.0     2.0      35.0   

    pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha dec_o  attr_o  sinc_o  \
0       20.0      20.0      20.0       0.0       5.0   Non     6.0     8.0   
10      20.0      20.0      20.0       0.0       5.0   Non     8.0     7.0   
20      20.0      20.0      20.0       0.0       5.0   Non     7.0     8.0   
30      20.0      20.0      20.0       0.0       5.0   Non     6.0     7.0   
40      20.0      20.0      20.0       0.0       5.0   Non     6.0     8.0   

    intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age      field  \
0       8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0        Law   
10      6.0    9.0    7.0     4.0     7.0     2.0    2.0  24.0        law   
20      6.0    5.0    8.0     4.0     5.0     2.0    1.0  25.0  Economics   
30      8.0    7.0    7.0     5.0     6.0     3.0    2.0  23.0        Law   
40      8.0    8.0    7.0     6.0     6.0     5.0    2.0  21.0        Law   

    field_cd undergra mn_sat tuition  race  imprace  imprelig  \
0        1.0      NaN    NaN     NaN   4.0      2.0       4.0   
10       1.0      NaN    NaN     NaN   2.0      2.0       5.0   
20       2.0      NaN    NaN     NaN   2.0      8.0       4.0   
30       1.0      NaN    NaN     NaN   2.0      1.0       1.0   
40       1.0      NaN    NaN     NaN   2.0      8.0       1.0   

               from zipcode     income  goal  date  go_out  \
0           Chicago  60,521  69,487.00   2.0   7.0     1.0   
10          Alabama  35,223  65,929.00   1.0   5.0     1.0   
20      Connecticut   6,268        NaN   6.0   3.0     1.0   
30            Texas  77,096  37,754.00   1.0   5.0     1.0   
40  Bowdoin College  94,022  86,340.00   2.0   4.0     1.0   

                career  career_c  sports  tvsports  exercise  dining  museums  \
0               lawyer       NaN     9.0       2.0       8.0     9.0      1.0   
10                 law       NaN     3.0       2.0       7.0    10.0      8.0   
20           Economist       NaN     3.0       8.0       7.0     8.0      5.0   
30  lawyer/policy work       1.0     1.0       1.0       6.0     7.0      6.0   
40              lawyer       1.0     7.0       4.0       7.0     7.0      6.0   

    art  hiking  gaming  clubbing  reading   tv  theater  movies  concerts  \
0   1.0     5.0     1.0       5.0      6.0  9.0      1.0    10.0      10.0   
10  6.0     3.0     5.0       8.0     10.0  1.0      9.0     8.0       7.0   
20  5.0     8.0     4.0       5.0      7.0  8.0      7.0     7.0       7.0   
30  7.0     7.0     5.0       7.0      7.0  7.0      9.0     7.0       8.0   
40  8.0     6.0     6.0       8.0      6.0  8.0      6.0     6.0       3.0   

    music  shopping  yoga  exphappy  expnum  attr1_1  sinc1_1  intel1_1  \
0     9.0       8.0   1.0       3.0     2.0     15.0     20.0      20.0   
10    8.0       3.0   1.0       4.0     5.0     45.0      5.0      25.0   
20    5.0       8.0   7.0       4.0     2.0     35.0     10.0      35.0   
30    7.0       1.0   8.0       1.0     2.0     20.0     20.0      20.0   
40    7.0       8.0   3.0       7.0    10.0     20.0      5.0      25.0   

    fun1_1 

In [124]:
# configuration d'un code couleur pour Woman et Man
color_discrete_map = {
    'Man': 'blue',  
    'Woman': 'red'  
}
# Répartition Femme et Homme
id_unique = df.groupby('gender')['iid'].nunique().reset_index()

fig=px.bar(id_unique, x="gender", y="iid", color="gender", text="iid", color_discrete_map=color_discrete_map, title="Nombre de Femme et d'Homme participant à cette étude")
fig.show()


fig2 = px.histogram(df_unique, x="age",color="gender",
                    marginal="box",
                    color_discrete_map=color_discrete_map,
                    title="Répartition de l'age des candidats"
                    )
fig2.show()

In [125]:
# Quelques données liés à l'age de nos candidats
mask_age = (df_unique["age"] >= 23) & (df_unique["age"] <= 29)
df_unique_age = df_unique[mask_age]
distribution = df_unique_age["age"].count()
pourcentage_distribution = distribution*100/df_unique["age"].count()
print("L'age des candidats est compris entre 23 et 29 ans:", distribution )
print(f"Soit {pourcentage_distribution.round(2)}%")
mask_gender = df_unique["gender"] == "Woman"
df_unique_wgender = df_unique[mask_gender]
woman_mean = df_unique_wgender["age"].mean()
df_unique_mgender= df_unique[~mask_gender]
man_mean = df_unique_mgender["age"].mean()
print(f"L'age moyen des femmes est de {woman_mean.round(1)} ans, et l'age moyen des hommes est de {man_mean.round(1)} ans")


L'age des candidats est compris entre 23 et 29 ans: 377
Soit 69.43%
L'age moyen des femmes est de 26.1 ans, et l'age moyen des hommes est de 26.6 ans


#### La distribution de l'age montre que 2/3 des candidats sont agés entre 23 et 29 ans et qu'il y a quelques outliers pour les 2 partis. 
#### Les hommes présentent un age avec une distribution plus grande que les femmes et donc un age moyen plus élevé. Quand aux femmes elles couvrent principalement une plage d'age plus petit mais a la candidate la plus agée.

In [126]:
# Modification de la colonne "goal" pourune meilleur lecture des informations
df_unique.loc[:,"goal"] = df_unique["goal"].apply(lambda x: "Seemed like a fun night out" if x==1
	                                                            else "To meet new people" if x==2
                                                                else "To get a date" if x==3
                                                                else "Looking for a serious relationship" if x==4
                                                                else "To say I did it" if x==5
                                                                else "Other" if x==6
                                                                else x
).astype(str)

/tmp/ipykernel_1317/1849637918.py:2: FutureWarning:

Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '['To meet new people' 'Seemed like a fun night out' 'Other'
 'Seemed like a fun night out' 'To meet new people'
 'Seemed like a fun night out' 'Seemed like a fun night out'
 'Seemed like a fun night out' 'Seemed like a fun night out'
 'To meet new people' 'Seemed like a fun night out'
 'Seemed like a fun night out' 'To meet new people' 'To meet new people'
 'Seemed like a fun night out' 'Seemed like a fun night out'
 'To get a date' 'Seemed like a fun night out'
 'Looking for a serious relationship' 'Seemed like a fun night out'
 'Other' 'Seemed like a fun night out' 'Seemed like a fun night out'
 'To meet new people' 'Seemed like a fun night out'
 'Seemed like a fun night out' 'Seemed like a fun night out' 'nan'
 'Seemed like a fun night out' 'Seemed like a fun night out'
 'Seemed like a fun night out' 'Seemed like a fun night out'

In [127]:
fig_goal1=px.pie(df_unique, names="goal", title="Qu'attendent les candidats en participant à cette évènement?")

fig_goal2 = px.histogram(df_unique,x="goal", color="gender", color_discrete_map=color_discrete_map, barmode="group", title="Répartition par genre")

fig_goal1.show()
fig_goal2.show()

#### On constate qu'il y a une forte volonté des 2 coté de rencontrer de nouvelle personne et de passer une bonne soirée. On est dans l'essence même de Tinder faire rencontrer les gens.
#### L'objetcif d'avoir un Date ou une relation serieuse ne touche que 11%, et c'est principalement les hommes qui ont cette objectif.

In [128]:
# Taux de match de la soirée
# création d'un code couleur pour oui/non 
color_pie = {
    'Oui': 'green',  
    'Non': 'black',
} 
df_match=df.copy()
df_match['match'] = df_match['match'].replace({1: 'Oui',0: 'Non'}) 
fig=px.pie(df_match, names='match',color='match', title="Taux de match de la soirée", color_discrete_map=color_pie)
fig.show()

#### le Taux de match ne s'élève qu'a 16.5% ce qui est faible.


## Apprenons en plus sur nos candidats

In [129]:
# modifications des colonnes "career_c" et "field_c"
df_unique.loc[:,"field_cd"]=df_unique["field_cd"].apply(lambda x: "Law" if x==1
                                                else "Match" if x==2
                                                else "Social Science, Psychologist" if x==3 
                                                else "Medical Science, Pharmaceuticals, and Bio Tech" if x==4 
                                                else "Engineering" if x==5  
                                                else "English/Creative Writing/ Journalism" if x==6 
                                                else "History/Religion/Philosophy" if x==7 
                                                else "Business/Econ/Finance" if x==8
                                                else "Education, Academia" if x==9
                                                else "Biological Sciences/Chemistry/Physics" if x==10
                                                else "Social Work" if x==11
                                                else "Undergrad/undecided" if x==12
                                                else "Political Science/International Affairs" if x==13
                                                else "Filmif" if x==14
                                                else "Fine Arts/Arts Administration" if x==15
                                                else "Languages" if x==16
                                                else "Architecture" if x==17
                                                else "Other" if x==18
                                                else x
                                                ).astype(str)
df_unique.loc[:,"career_c"]=df_unique["career_c"].apply(lambda x: "Lawyer" if x==1 
                                                else "Academic/Research" if x==2
                                                else "Psychologist" if x==3
                                                else "Doctor/Medicine" if x==4
                                                else "Engineer" if x==5
                                                else "Creative Arts/Entertainment" if x==6
                                                else "Banking/Consulting/Finance/Marketing/Business/CEO/Entrepreneur/Admin" if x==7
                                                else "Real Estate" if x==8
                                                else "International/Humanitarian Affairs" if x==9
                                                else "Undecided" if x==10
                                                else "Social Work" if x==11
                                                else "Speech Pathology" if x==12
                                                else "Politics" if x==13
                                                else "Pro sports/Athletics" if x==14
                                                else "Other" if x==15
                                                else "Journalism" if x==16
                                                else "Architecture" if x==17
                                                else x
).astype(str)

/tmp/ipykernel_1317/84121120.py:2: FutureWarning:

Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '['Law' 'Law' 'Match' 'Law' 'Law' 'Law' 'Law'
 'Political Science/International Affairs'
 'Political Science/International Affairs'
 'Political Science/International Affairs' 'Business/Econ/Finance' 'Law'
 'Law' 'Law' 'Law' 'Business/Econ/Finance'
 'Political Science/International Affairs' 'Business/Econ/Finance'
 'Business/Econ/Finance' 'Business/Econ/Finance' 'Engineering'
 'Education, Academia' 'Social Science, Psychologist' 'Social Work'
 'Social Work' 'Social Work' 'Social Science, Psychologist'
 'Social Science, Psychologist' 'Social Work' 'Law' 'Law'
 'Social Science, Psychologist' 'Social Science, Psychologist'
 'Business/Econ/Finance' 'Match' 'Match' 'Social Science, Psychologist'
 'Law' 'Law' 'nan' 'Law' 'Engineering' 'Business/Econ/Finance'
 'Social Science, Psychologist' 'Business/Econ/Finance'
 'Business/Econ/Finance' 'Bus

In [130]:

# Compter les occurrences de chaque 'field_cd' par genre
df_count = df_unique.groupby(['field_cd', 'gender']).size().reset_index(name='counts')

# Trier les valeurs par 'counts' de manière croissante
df_count = df_count.sort_values(by='counts', ascending=True)

fig = px.bar(
    df_count,
    y="field_cd",
    x="counts",
    color="gender",
    color_discrete_map=color_discrete_map,
    barmode="group",
    title="Études des participants par genre",
    orientation='h'  
)

# Affichage de la figure
fig.show()

#### Les femmes ont réalisé des études plus porté sur le social, néanmoins on constate une plus grande diversité d'étude et de domaine. Quand aux hommes les etudes éconimique et finance sont nettement privilégiées, vient juste après des études d'ingenieurs et de science. Les hommes sont très peu présent dans les autres domaines dont le social. 

In [131]:

df_count = df_unique.groupby(['career_c', 'gender']).size().reset_index(name='counts')

# Trier les valeurs par 'counts' de manière croissante
df_count = df_count.sort_values(by='counts', ascending=True)

fig = px.bar(
    df_count,
    y="career_c",
    x="counts",
    color="gender",
    color_discrete_map=color_discrete_map,
    barmode="group",
    title="Activité professionnelle des participants par genre",
    orientation='h'  
)

# Affichage de la figure
fig.show()

#### On retrouve chez les femmes cette diversité dans l'emploi avec le domaine de la recherche où elles sont les plus présente suivi du secteur commerce et finance qui lui est très important chez les hommes. On retrouve chez les femmes l'intêret du social dans leur activité professionnelle.

In [132]:
# modification des colonnes "race" pour une meilleur lisibilité
df_unique.loc[:, "race"] = df_unique["race"].apply(lambda x: "Black/African" if x==1 
                                                         else "Caucasian/European" if x==2
                                                         else "Latino/Hispanic" if x==3
                                                         else "Asian" if x==4
                                                         else "Native American" if x==5
                                                         else "Other" if x==6
                                                         else x).astype(str)

df_unique.loc[:,"imprace"] = df_unique["imprace"].apply(lambda x: "Not important" if x==0
                                                        else "Not important" if x==1
                                                        else "Not important" if x==2
                                                        else "Not important" if x==3
                                                        else "Not important" if x==4
                                                        else "Important" if x==5
                                                        else "Important" if x==6
                                                        else "Important" if x==7
                                                        else "Very important" if x==8
                                                        else "Very important" if x==9
                                                        else "Very important" if x==10
                                                        else x).astype(str)

/tmp/ipykernel_1317/2446832980.py:2: FutureWarning:

Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '['Asian' 'Caucasian/European' 'Caucasian/European' 'Caucasian/European'
 'Caucasian/European' 'Asian' 'Asian' 'Caucasian/European' 'Other'
 'Caucasian/European' 'Caucasian/European' 'Caucasian/European' 'Asian'
 'Caucasian/European' 'Latino/Hispanic' 'Caucasian/European'
 'Caucasian/European' 'Caucasian/European' 'Caucasian/European'
 'Caucasian/European' 'Asian' 'Black/African' 'Asian' 'Caucasian/European'
 'Caucasian/European' 'Caucasian/European' 'Latino/Hispanic'
 'Caucasian/European' 'Caucasian/European' 'Caucasian/European'
 'Caucasian/European' 'Asian' 'Caucasian/European' 'Asian'
 'Caucasian/European' 'Caucasian/European' 'Caucasian/European'
 'Black/African' 'Latino/Hispanic' 'Asian' 'Asian' 'Asian'
 'Caucasian/European' 'Caucasian/European' 'Caucasian/European'
 'Caucasian/European' 'Caucasian/European' 'Caucasian/European

In [133]:
fig = px.histogram(
    df_unique,
    x="race",
    color="gender",
    color_discrete_map=color_discrete_map,
    title="Répartition par race et genre",
    labels={"race_o": "Race", "gender": "Genre"},
    barmode='group' 
)
fig.show()

In [134]:
fig = px.histogram(df_unique, 
                y="imprace", 
                color="gender",
                color_discrete_map=color_discrete_map,
                title="Importance de la race par genre",
                labels={"imprace": "Race", "gender": "Genre"},
                barmode='group' 
)
fig.show()

#### La race n'est pas un critère primordial pour nos candidats, néanmoins 1/3 d'entre eux n'y sont pas insensible et se sont les femmes qui y porteraient le plus attention. 

#### Vérifions si cela se traduit par plus de OUI a un second rendez vous chez les femmes dans ce cas là.

In [135]:
# On filtre les données pour les cas où 'samerace' est Oui
df_same_race = df[df['samerace']=="Oui"]

fig = px.histogram(
    df_same_race,
    x="dec",
    color="gender",
    color_discrete_map=color_discrete_map,
    title="Décision d'une seconde rencontre par genre pour la même race",
    labels={"dec": "Décision", "gender": "Genre"},
    barmode='group'
)

fig.show()


#### Les hommes ont tendance à dire plus souvent Oui a un prochain rendez vous chez les partenaires de même race que les femmes.

#### Confirmons cela en analysons les resultats dans le cas où ils ne sont pas de même race 

In [136]:
# Filtrer les données pour les cas où 'samerace' est Non
df_same_race = df[df['samerace']=="Non"]

fig = px.histogram(
    df_same_race,
    x="dec",
    color="gender",
    color_discrete_map=color_discrete_map,
    title="Décision d'une seconde rencontre par genre n'ayant pas la même race",
    labels={"dec": "Décision", "gender": "Genre"},
    barmode='group'
)

fig.show()

#### Les résultats nous montrent là aussi une tendance à dire oui pour une second rendez vous chez les hommes même si leur partenaire n'est pas de même race.
#### On ne peut donc pas avancer avec certitude l'hypothèse que la race est un facteur déterminant chez les hommes dans leur prise de décision à un second rendez-vous.
#### Pour finir regardons notre taux de match et si celui ci est plus important dans le cas où les candidats sont d'origine raciale commune.



In [137]:
# voyons maintenant si le taux de match est plus important si le partenaire est de même race
# configuration d'un code couleur
color_match={'Même origine':'green','Origine différente':'black'}

df_race_match = df.groupby('samerace', as_index=False)['match'].mean()
df_race_match['samerace'] = df_race_match['samerace'].replace({"Oui": 'Origine différente', "Non": 'Même origine'})

fig1 = px.bar(df_race_match, x='samerace', y='match',
              title="Taux de match selon origine raciale commune",
              labels={'match': 'Taux de match', 'samerace': 'Origine raciale'},
              color='samerace', text='match', color_discrete_map=color_match)
fig1.update_traces(texttemplate='%{text:.2%}', textposition='inside')
fig1.update_layout(yaxis_tickformat=".0%", showlegend=False)

fig1.show()

#### Le resultat ne montre pas que l'origine raciale favorise les matchs, il n'est donc pas un critère important lors d'une rencontre.
#### Faisons un point rapide sur la religion et son importance pour les candidats.

In [138]:
# modification des colonnes race pour une meilleur lisibilité
df_unique.loc[:,"imprelig"] = df_unique["imprelig"].apply(lambda x: "Not important" if x==0
                                                            else "Not important" if x==1
                                                            else "Not important" if x==2
                                                            else "Not important" if x==3
                                                            else "Not important" if x==4
                                                            else "Important" if x==5
                                                            else "Important" if x==6
                                                            else "Important" if x==7
                                                            else "Very Important" if x==8
                                                            else "Very Important" if x==9
                                                            else "Very Important" if x==10
                                                            else x).astype(str)

/tmp/ipykernel_1317/709554979.py:2: FutureWarning:

Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '['Not important' 'Important' 'Not important' 'Not important'
 'Not important' 'Not important' 'Not important' 'Not important'
 'Not important' 'Not important' 'Not important' 'Not important'
 'Important' 'Not important' 'Not important' 'Not important'
 'Not important' 'Very Important' 'Not important' 'Not important'
 'Very Important' 'Not important' 'Important' 'Not important'
 'Not important' 'Not important' 'Important' 'nan' 'Important'
 'Not important' 'Very Important' 'Not important' 'Very Important'
 'Not important' 'Important' 'Very Important' 'Very Important'
 'Not important' 'Not important' 'Important' 'Not important' 'Important'
 'Not important' 'Not important' 'Important' 'Not important'
 'Not important' 'Not important' 'Not important' 'Not important'
 'Very Important' 'Very Important' 'Not important' 'Not important'
 'Not 

In [139]:
fig = px.histogram(
    df_unique,
    x="imprelig",
    color="gender",
    color_discrete_map=color_discrete_map,
    title="Importance de la religion par genre",
    labels={"imprelig": "Religion", "gender": "Genre"},
    barmode='group' 
)
fig.show()

#### La religion n'est pas importante pour les 2 parties. Néanmoins dans le cas où celle ci commence à avoir de l'importance, se sont les hommes qui y portent le plus attention.

#### Nos dernières analyses nous permettent de commencer à démontrer qu'il semblerait que les hommes ont été plutot favorable à un second rendez vous vis a vis des femmes. Nous allons essayer de confirmer ce point.

In [140]:
#combien de personne on dit oui a un 2 eme rendez vous
fig = make_subplots(rows=1, cols=3, subplot_titles=("fig 1", "fig 2", "fig 3"))

fig1 = px.pie(df, names="dec",color="dec", color_discrete_map=color_pie, title="Répartition des décisions")
fig1.show()


# Création des sous-graphiques
fig = make_subplots(rows=1, cols=2, subplot_titles=("Répartition des 'Oui' par genre", "Répartition des 'Non' par genre"))

# Filtrer les données pour les 'Oui'
df_yes = df[df["dec"] == "Oui"]
# Ajout du deuxième graphique (histogramme pour 'Oui')
hist_yes = px.histogram(df_yes, x="gender", color="gender", color_discrete_map=color_discrete_map)
for trace in hist_yes.data:
    fig.add_trace(trace, row=1, col=1)

# Filtrer les données pour les 'Non'
df_no = df[df["dec"] == "Non"]
# Ajout du troisième graphique (histogramme pour 'Non')
hist_no = px.histogram(df_no, x="gender", color="gender", color_discrete_map=color_discrete_map)
for trace in hist_no.data:
    trace.showlegend = False
    fig.add_trace(trace, row=1, col=2)

# Mise à jour des axes et du layout
fig.update_layout(
    title_text="Analyse des décisions",
    font=dict(family="Arial", size=12, color="Black")
)
fig.show()

#### Nous constatons qu'il y a pratiquement 2/3 de Non pour un second rendez vous. Et que se sont les hommes qui ont dit le plus OUI, ce qui confirme notre hypothèse précédente.
#### Maintenant voyons quelles sont les attributs les plus importants recherchés chez un partenaire


# Importance des attributs
#### Lexiques des attributs:
####        - attr => Attirance	
####        - sinc => Sincère		
####        - intel => Intelligent		
####        - fun= > Drole/Amusant			
####        - amb => Ambitieux		
####        - shar => Intérêts commun
# Commençons par faire un état avant les Dates. 
#### Pour cela nous devront faire attention aux différentes vagues de Dates. En effet le document word nous indique que la vague de Dates 6 à 9 a noté les critères attributs de 1 à 10 quand au autres vagues ils ont réparti 100 points sur les différents critères. Cela concerne 3 questions, nous devrons donc faire 2 analyses en parallèle
#### Vague 1 à 5 et 10 à 21 répartition de 100 points
#### Vague 6 à 9 note sur 10



In [141]:
# Nous créons un dataset avec la vague 6 à 9 pour séparer les vagues
mask_vague = (df['iid'] >= 6) & (df['iid'] <= 9)
df_vague = df[mask_vague]
df_vague.describe(include='all')

iid         id gender        idg  condtn  wave  round  \
count   40.000000  40.000000     40  40.000000    40.0  40.0   40.0   
unique        NaN        NaN      1        NaN     NaN   NaN    NaN   
top           NaN        NaN  Woman        NaN     NaN   NaN    NaN   
freq          NaN        NaN     40        NaN     NaN   NaN    NaN   
mean     7.500000   7.500000    NaN  14.000000     1.0   1.0   10.0   
std      1.132277   1.132277    NaN   2.264554     0.0   0.0    0.0   
min      6.000000   6.000000    NaN  11.000000     1.0   1.0   10.0   
25%      6.750000   6.750000    NaN  12.500000     1.0   1.0   10.0   
50%      7.500000   7.500000    NaN  14.000000     1.0   1.0   10.0   
75%      8.250000   8.250000    NaN  15.500000     1.0   1.0   10.0   
max      9.000000   9.000000    NaN  17.000000     1.0   1.0   10.0   

         position  positin1      order    partner        pid      match  \
count   40.000000       0.0  40.000000  40.000000  40.000000  40.000000   
unique        NaN       NaN        NaN        NaN        NaN        NaN   
top           NaN       NaN        NaN        NaN        NaN        NaN   
freq          NaN       NaN        NaN        NaN        NaN        NaN   
mean     6.000000       NaN   5.500000   5.500000  15.500000   0.475000   
std      3.434366       NaN   2.908872   2.908872   2.908872   0.505736   
min      1.000000       NaN   1.000000   1.000000  11.000000   0.000000   
25%      4.000000       NaN   3.000000   3.000000  13.000000   0.000000   
50%      6.500000       NaN   5.500000   5.500000  15.500000   0.000000   
75%      8.500000       NaN   8.000000   8.000000  18.000000   1.000000   
max     10.000000       NaN  10.000000  10.000000  20.000000   1.000000   

         int_corr samerace      age_o    race_o    pf_o_att   pf_o_sin  \
count   40.000000       40  40.000000  40.00000   40.000000  40.000000   
unique        NaN        2        NaN       NaN         NaN        NaN   
top           NaN      Non        NaN       NaN         NaN        NaN   
freq          NaN       30        NaN       NaN         NaN        NaN   
mean     0.079000      NaN  25.200000   2.30000   44.233000   7.911000   
std      0.321238      NaN   2.593878   0.64847   22.086789   7.656691   
min     -0.510000      NaN  22.000000   2.00000   19.000000   0.000000   
25%     -0.215000      NaN  23.000000   2.00000   30.000000   0.000000   
50%      0.110000      NaN  24.500000   2.00000   35.000000   7.500000   
75%      0.335000      NaN  27.000000   2.00000   50.000000  15.000000   
max      0.680000      NaN  30.000000   4.00000  100.000000  20.000000   

         pf_o_int   pf_o_fun   pf_o_amb   pf_o_sha dec_o     attr_o  \
count   40.000000  40.000000  40.000000  40.000000    40  40.000000   
unique        NaN        NaN        NaN        NaN     2        NaN   
top           NaN        NaN        NaN        NaN   Oui        NaN   
freq          NaN        NaN        NaN        NaN    27        NaN   
mean    16.511000  16.911000   4.511000   9.922000   NaN   7.475000   
std      9.783822  12.760568   5.222642   7.360794   NaN   1.358497   
min      0.000000   0.000000   0.000000   0.000000   NaN   5.000000   
25%     11.110000  10.000000   0.000000   5.000000   NaN   6.000000   
50%     19.500000  10.555000   2.500000  10.000000   NaN   7.500000   
75%     25.000000  20.000000  10.000000  15.000000   NaN   8.000000   
max     30.000000  40.000000  14.000000  22.220000   NaN  10.000000   

           sinc_o    intel_o      fun_o      amb_o     shar_o     like_o  \
count   40.000000  40.000000  40.000000  39.000000  40.000000  40.000000   
unique        NaN        NaN        NaN        NaN        NaN        NaN   
top           NaN        NaN        NaN        NaN        NaN        NaN   
freq          NaN        NaN        NaN        NaN        NaN        NaN   
mean     7.650000   8.100000   7.425000   7.461538   6.525000   7.300000   
std      1.442043   1.373747   1.583367   1.536335   1.921371 

#### En faisant un simple describe pour regarder si notre nouveau dataset est correctement filtré nous constatons que les colonnes concernées par une notation sur 10 ont été noté sur le même principe que les autres vagues c'est à dire une répartition des points sur 100. Donc il n'est pas necessaire de diviser notre dataset en 2.

In [142]:

# Que cherche une personne chez le sexe opposé avant les Dates
df_attr = df_unique[['attr1_1','sinc1_1','intel1_1','fun1_1','amb1_1','shar1_1','gender']]
moyenne_avant = (df_attr.groupby('gender').mean().round(2))
display(moyenne_avant)
df_attr = df_attr.drop('gender', axis=1)
moyennes = df_attr.mean().sort_values()
fig = px.bar(
    x=moyennes.index,
    y=moyennes.values,
    color=moyennes.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title='Moyenne des résultats sur les attributs recherchés chez le sexe opposé avant les Dates', 
    color_continuous_scale="GnBu"
)
fig.show()

attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1
gender                                                     
Man       27.25    16.38     19.39    17.6    8.82    10.98
Woman     18.02    18.22     20.97    17.3   12.82    12.70

In [143]:
# que pensez vous que la plupart de vos collègues hommes / femmes recherchent dans le sexe opposé
df_attr4 = df_unique[['attr4_1','sinc4_1','intel4_1','fun4_1','amb4_1','shar4_1', 'gender']]
display(df_attr4.groupby('gender').mean().round(2))
df_attr4 = df_attr4.drop('gender', axis=1)
moyennes = df_attr4.mean().sort_values()
fig = px.bar(
    x=moyennes.index,
    y=moyennes.values,
    color=moyennes.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title='Que pensez vous que recherche vos collègues hommes/femmes dans le sexe opposé',
    color_continuous_scale="GnBu"
)
fig.show()

attr4_1  sinc4_1  intel4_1  fun4_1  amb4_1  shar4_1
gender                                                     
Man       28.31    10.77     12.05   16.14    7.71    10.84
Woman     24.35    11.39     13.05   15.07   11.85    11.17

#### Cette fois ci les 2 partis sont d'accord et pour eux le sexe opposé cherche l'attirance, qui est nettement devant. Puis quelqu'un de "drole/amusant".
#### L'ambition et les intérêts commun ne sont toujours pas péblicités

In [144]:
#Que pensez-vous que le sexe opposé recherche dans un rendez-vous
df_attr2 = df_unique[['attr2_1','sinc2_1','intel2_1','fun2_1','amb2_1','shar2_1','gender']]
display(df_attr2.groupby('gender').mean().round(2))
df_attr2 = df_attr2.drop('gender', axis=1)
moyennes = df_attr2.mean().sort_values()
fig = px.bar(
    x=moyennes.index,
    y=moyennes.values,
    color=moyennes.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title='Que pensez vous que recherche le sexe opposé dans un rendez vous',
    color_continuous_scale="GnBu"
)
fig.show()

attr2_1  sinc2_1  intel2_1  fun2_1  amb2_1  shar2_1
gender                                                     
Man       25.01    15.13     16.33   18.02   14.36    11.16
Woman     35.60    11.28     12.48   19.05    9.11    12.63

#### Là aussi les 2 partis sont en phase avec le fait que lors d'un Date le sexe opposé recherche l'attirance en premier lieu puis une personne "drole/amusante". On notera que les femmes ont nettement mise en avant ce critère concidérant donc que les hommes s'attachent énormement au physique.
#### Une fois de plus l'ambition et les intérêts communs ferment la marche.

## Etat à mis chemin de la soirée du Date

In [145]:
# résultat le soir même à mi chemin
df_attr1_s = df_unique[['attr1_s','sinc1_s','intel1_s','fun1_s','amb1_s','shar1_s', 'gender']]
display(df_attr1_s.groupby('gender').mean())
df_attr1_s=df_attr1_s.drop('gender', axis=1)
moyennes = df_attr1_s.mean().sort_values()
fig = px.bar(
    x=moyennes.index,
    y=moyennes.values,
    color=moyennes.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title='Le soir même à mi chemin: Moyenne des résultats sur les attributs recherchés chez le sexe opposé',
    color_continuous_scale="GnBu"
)
fig.show()

attr1_s    sinc1_s   intel1_s     fun1_s     amb1_s    shar1_s
gender                                                                  
Man     23.371241  14.508345  16.147034  14.993172   9.483724  10.897172
Woman   16.669559  15.918824  17.336397  14.729706  12.194559  13.100147

#### Nous pouvons constater que les 2 partis ont maintenu leur ligne de conduite dans l'importance des critères et donc leur prise de décision. Pour les hommes c'est l'attirance suivi de l'intelligence et pour les femmes l'intelligence reste le critère le plus important mais tout juste devant l'attirance qui gagne une place dans l'importance des attributs.
#### Voyons si cela c'est traduit concrétement dans leur prise de décision.

In [146]:
# Répartitions des attributs le soir même avec une décision à OUI
df_dec_yes= df[df['dec'] == "Oui"]
df_dec_yes = df_dec_yes[['attr','sinc','intel','fun','amb','shar','gender']]
yes=df_dec_yes.groupby("gender").mean()
display(yes)
df_dec_yes=df_dec_yes.drop("gender", axis=1)
moyennes_yes = df_dec_yes.mean()
fig =px.bar(
    x=moyennes_yes.index,
    y=moyennes_yes.values,
    color=moyennes_yes.values,
    labels={'x': 'Colonnes', 'y': 'Moyenne de la répartiton sur 10'},
    title="Moyenne des résultats pour une décision d'une nouvelle rencontre à OUI",
    color_continuous_scale="GnBu"
)
fig.show()

attr      sinc     intel       fun       amb      shar
gender                                                            
Man     7.448739  7.568112  7.621498  7.296134  6.995148  6.371681
Woman   7.072787  7.632566  7.932164  7.381710  7.374647  6.583885

#### Lors de l'évaluation des attributs de chaque partenaires rencontré, on constate que le critère attirance arrive en 4 eme position lorsque la personne accepte un second rendez vous. Cependant la note est tout aussi élevé et très proche que les 3 premiers soit une note au delà de 7.
#### Les femmes restent fidèles à leurs critères dans leurs decisions, quand aux critères des hommes ils sont plus en phase avec ceux des femmes, même si l'attirance est en 3eme position quand pour la femme il arrive en 5 eme.

In [147]:
# Répartitions des attributs le soir même avec une décision à Non
df_dec_no= df[df['dec'] == "Non"]
df_dec_no = df_dec_no[['attr','sinc','intel','fun','amb','shar','gender']]
no = df_dec_no.groupby("gender").mean()
display(no)
df_dec_no=df_dec_no.drop("gender", axis=1)
moyennes_no = df_dec_no.mean()

fig =px.bar(
    x=moyennes_no.index, 
    y=moyennes_no.values,
    color=moyennes_no.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title="Moyenne des résultats pour une décision d'une nouvelle rencontre à NON",
    color_continuous_scale="GnBu"
)
fig.show()

attr      sinc     intel       fun       amb      shar
gender                                                            
Man     5.531739  6.952563  6.977426  5.798275  6.237808  4.747230
Woman   5.231626  6.780189  7.157363  5.613764  6.704489  4.695865

In [148]:
moyennes_no=pd.DataFrame(moyennes_no).T
moyennes_yes=pd.DataFrame(moyennes_yes).T
print(moyennes_no)
print(moyennes_yes)

       attr      sinc     intel      fun       amb      shar
0  5.367316  6.858039  7.076063  5.69786  6.493288  4.719353
       attr      sinc     intel       fun       amb      shar
0  7.285258  7.596264  7.756621  7.333575  7.159431  6.462741


In [149]:
# les principaux écarts négatif vs le Oui
ecart_attra = (((moyennes_no.iloc[0, 0]*100/moyennes_yes.iloc[0, 0])-100).round(2))
print(f"Dans le cas d'un NON l'attirance baisse de {ecart_attra} % par rapport au OUI")
ecart_share = (((moyennes_no.iloc[0, 5]*100/moyennes_yes.iloc[0, 5])-100).round(2))
print(f"Dans le cas d'un NON l'interet commun baisse de {ecart_share} % par rapport au OUI")
ecart_fun = (((moyennes_no.iloc[0, 3]*100/moyennes_yes.iloc[0, 3])-100).round(2))
print(f"Dans le cas d'un NON le drole/amusant baisse de {ecart_fun} % par rapport au OUI")


Dans le cas d'un NON l'attirance baisse de -26.33 % par rapport au OUI
Dans le cas d'un NON l'interet commun baisse de -26.98 % par rapport au OUI
Dans le cas d'un NON le drole/amusant baisse de -22.3 % par rapport au OUI


#### Dans le cas du refus on retrouve l'intelligence et la sincérité leader des critères mais légèrement moins bien noté que dans le cas d'un OUI ( baisse de 10% environ). L'attirance arrive en 5 eme position avec une note lègèrement supérieur à 5 soit -26.33% par rapport à sa note du OUI. Nous avons aussi les intérêts commun et le critère "drôle/amusant" qui ont subit de grosse chute. Nous retrouvons donc bien l'attirance et le " drôle/amusant" comme critère important pour les candiats dans leur décision conforme aux réponses apportées lors des questions " que cherchez vous dans le sexe opposé". Cependant apparait aussi les intérêts en communs qui jusqu'à présent ne semblaient pas être un facteur influençant fortement le choix de nos candidats. Hors il subit la plus grosse chute dans le cas d'un refus pour un second rendez vous avec -26.98%. Il semblerait que ce critère ait un impact signifiant tout compte fait, gardons ce point en tête.

## Analyse des résultats au lendemain.

In [150]:
# résultat le soir même à mi chemin
df_attr1_2 = df_unique[['attr1_2','sinc1_2','intel1_2','fun1_2','amb1_2','shar1_2', 'gender']]
display(df_attr1_2.groupby('gender').mean())
df_attr1_2=df_attr1_2.drop('gender', axis=1)
moyennes = df_attr1_2.mean().sort_values()
fig = px.bar(
    x=moyennes.index,
    y=moyennes.values,
    color=moyennes.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title='Le lendemain: Moyenne des résultats sur les attributs recherchés chez le sexe opposé',
    color_continuous_scale="GnBu"
)
fig.show()

attr1_2    sinc1_2   intel1_2     fun1_2     amb1_2    shar1_2
gender                                                                  
Man     30.126721  15.204534  16.751457  17.736356   8.535385  11.512024
Woman   21.862996  16.576639  19.118109  17.489874  11.446849  13.942521

Les femmes privilégies plus l'attirance même si l'intelligence reste juste derrière. Les hommes n'ont pas changé leur position et ont même accentué l'intérêt de ce critère qui maintenant surpasse tous les autres avec 50% du poids des critères. 

In [151]:
# Répartitions des points donnés aux attributs dans la prise de decision
df_attr7 = df[['attr7_2','sinc7_2','intel7_2','fun7_2','amb7_2','shar7_2','gender']]
display(df_attr7.groupby('gender').mean())
df_attr7=df_attr7.drop('gender', axis=1)
moyennes = df_attr7.mean().sort_values()
fig = px.bar(
    x=moyennes.index,
    y=moyennes.values,
    color=moyennes.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title='Analyse au lendemain : Répartition moyenne des points donnés aux attributs influençant la decision',
    color_continuous_scale="GnBu",
)
fig.show()



attr7_2    sinc7_2   intel7_2     fun7_2    amb7_2    shar7_2
gender                                                                 
Man     37.314525  11.684266  14.986416  19.342738  6.386484  10.394931
Woman   28.630964  15.263889  15.580331  18.426485  8.132937  13.779942

#### Le résultat est parlant des 2 cotés, c'est le critère d'attirance qui a influencé le choix d'un second rendez vous ou non. 
#### Voyons si l'ordre d'importance des critères ont changé entre avant et après les Dates.

In [152]:
# Question au lendemain : Que cherchez vous chez le sexe opposé? comparé à avant le Date
df_attr8 = df_unique[['attr1_2','sinc1_2','intel1_2','fun1_2','amb1_2','shar1_2','gender','attr1_1','sinc1_1','intel1_1','fun1_1','amb1_1','shar1_1']]
moyenne_genre=df_attr8.groupby('gender').mean()
display(moyenne_genre)
df_attr8=df_attr8.drop('gender', axis=1)
moyennes_lendemain = df_attr8.mean().sort_values()
# Créer une liste pour stocker les données du graphique
plot_data_list = []

# On réorganiser les données pour le graphique
prefixes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']
moments = ['Lendemain du Date', 'Avant le premier Date']  

for prefix in prefixes:
    for moment, suffix in zip(moments, ['1_2', '1_1']):
        col_name = f"{prefix}{suffix}"
        plot_data_list.append({
            'Attribute': prefix,
            'Moment': moment,
            'Value': moyennes_lendemain[col_name]
        })

# Créer le DataFrame pour le graphique
plot_data = pd.DataFrame(plot_data_list)


fig = px.bar(
    plot_data,
    x='Attribute',
    y='Value',
    color='Moment',
    barmode='group',
    labels={'Attribute': 'Attribut', 'Value': 'Moyenne'},
    title='Comparaison sur 2 périodes de la répartition moyenne des points donnés aux attributs influençant la décision',
    color_discrete_sequence=['blue', 'green']
)


fig.show()

attr1_2    sinc1_2   intel1_2     fun1_2     amb1_2    shar1_2  \
gender                                                                     
Man     30.126721  15.204534  16.751457  17.736356   8.535385  11.512024   
Woman   21.862996  16.576639  19.118109  17.489874  11.446849  13.942521   

          attr1_1   sinc1_1   intel1_1     fun1_1     amb1_1    shar1_1  
gender                                                                   
Man     27.248800  16.37960  19.387418  17.600839   8.823956  10.982271  
Woman   18.020372  18.22223  20.971004  17.299108  12.818476  12.697836

In [153]:
moyennes_lendemain=pd.DataFrame(moyennes_lendemain).T
moyennes_lendemain

amb1_2     amb1_1    shar1_1    shar1_2    sinc1_2    sinc1_1     fun1_1  \
0  9.964103  10.806476  11.832126  12.704722  15.877856  17.290754  17.451363   

      fun1_2   intel1_2   intel1_1    attr1_1    attr1_2  
0  17.615402  17.912825  20.170478  22.685478  26.080227

In [154]:
ecart_attra2 = (((moyennes_lendemain.iloc[0, 11]*100/moyennes_lendemain.iloc[0, 10])-100).round(2))
print(f"L'interet pour l'attirance évolue de {ecart_attra2} % au lendemain des Dates")
ecart_attra2_woman=(((moyenne_genre.iloc[1, 0]*100/moyenne_avant.iloc[1, 0])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_attra2_woman} %")
ecart_attra2_man=(((moyenne_genre.iloc[0, 0]*100/moyenne_avant.iloc[0, 0])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_attra2_man} %")

print("")

ecart_sinc= (((moyennes_lendemain.iloc[0, 4]*100/moyennes_lendemain.iloc[0, 5])-100).round(2))
print(f"L'interet pour la sincérité évolue de {ecart_sinc} % au lendemain des Dates")
ecart_sinc_woman=(((moyenne_genre.iloc[1, 1]*100/moyenne_avant.iloc[1, 1])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_sinc_woman} %")
ecart_sinc_man=(((moyenne_genre.iloc[0, 1]*100/moyenne_avant.iloc[0, 1])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_sinc_man} %")

print("")

ecart_intel= (((moyennes_lendemain.iloc[0, 8]*100/moyennes_lendemain.iloc[0, 9])-100).round(2))
print(f"L'interet pour l'intelligence évolue de {ecart_intel} % au lendemain des Dates")
ecart_intel_woman=(((moyenne_genre.iloc[1, 2]*100/moyenne_avant.iloc[1, 2])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_intel_woman} %")
ecart_intel_man=(((moyenne_genre.iloc[0, 2]*100/moyenne_avant.iloc[0, 2])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_intel_man} %")

print("")

ecart_fun= (((moyennes_lendemain.iloc[0, 7]*100/moyennes_lendemain.iloc[0, 6])-100).round(2))
print(f"L'interet pour le 'drole/amusant' évolue de {ecart_fun} % au lendemain des Dates")
ecart_fun_woman=(((moyenne_genre.iloc[1, 3]*100/moyenne_avant.iloc[1, 3])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_fun_woman} %")
ecart_fun_man=(((moyenne_genre.iloc[0, 3]*100/moyenne_avant.iloc[0, 3])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_fun_man} %")

print("")

ecart_amb= (((moyennes_lendemain.iloc[0, 0]*100/moyennes_lendemain.iloc[0, 1])-100).round(2))
print(f"L'interet pour l'ambition évolue de {ecart_amb} % au lendemain des Dates")
ecart_amb_woman=(((moyenne_genre.iloc[1, 4]*100/moyenne_avant.iloc[1, 4])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_amb_woman} %")
ecart_amb_man=(((moyenne_genre.iloc[0, 4]*100/moyenne_avant.iloc[0, 4])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_amb_man} %")

print("")

ecart_shar= (((moyennes_lendemain.iloc[0, 3]*100/moyennes_lendemain.iloc[0, 2])-100).round(2))
print(f"L'interet pour l'intérêt commun évolue de {ecart_shar} % au lendemain des Dates")
ecart_shar_woman=(((moyenne_genre.iloc[1, 5]*100/moyenne_avant.iloc[1, 5])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_shar_woman} %")
ecart_shar_man=(((moyenne_genre.iloc[0, 5]*100/moyenne_avant.iloc[0, 5])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_shar_man} %")

L'interet pour l'attirance évolue de 14.96 % au lendemain des Dates
L'ecart chez les femmes est de 21.33 %
L'ecart chez les hommes est de 10.56 %

L'interet pour la sincérité évolue de -8.17 % au lendemain des Dates
L'ecart chez les femmes est de -9.02 %
L'ecart chez les hommes est de -7.18 %

L'interet pour l'intelligence évolue de -11.19 % au lendemain des Dates
L'ecart chez les femmes est de -8.83 %
L'ecart chez les hommes est de -13.61 %

L'interet pour le 'drole/amusant' évolue de 0.94 % au lendemain des Dates
L'ecart chez les femmes est de 1.1 %
L'ecart chez les hommes est de 0.77 %

L'interet pour l'ambition évolue de -7.8 % au lendemain des Dates
L'ecart chez les femmes est de -10.71 %
L'ecart chez les hommes est de -3.23 %

L'interet pour l'intérêt commun évolue de 7.37 % au lendemain des Dates
L'ecart chez les femmes est de 9.78 %
L'ecart chez les hommes est de 4.85 %


#### L'attirance reste le critère le plus important et a subit une hausse significative de + 16.45%. C'est chez les femmes que la hausse est la plus importante avec + 21.48%. L'intelligence est le critère qui a subit la plus grosse baisse -12.1%, notament chez les hommes avec -14.1%. La sincérité aussi subit une baisse équilibré chez nos 2 partis avec -8.8%. Parcontre les intérêts commun semblent etre un critère qui prend un peu plus de poids en réalisant la 2eme plus forte hausse avec + 7.73% et c'est chez les femmes que ce critère à pris le plus de valeur.

### 3 à 4 semaines après

In [155]:
# Répartitions des points par importance des critères attendu
df_attr1_3 = df[['attr1_3','sinc1_3','intel1_3','fun1_3','amb1_3','shar1_3','gender']]
display(df_attr1_3.groupby('gender').mean())
df_attr1_3=df_attr1_3.drop('gender', axis=1)
moyennes = df_attr1_3.mean().sort_values()
fig = px.bar(
    x=moyennes.index,
    y=moyennes.values,
    color=moyennes.values,
    labels={'x': 'colonnes', 'y': 'Moyenne'},
    title='Analyse 3/4 semaines après : Répartition moyenne des points donnés aux attributs influençant la decision',
    color_continuous_scale="GnBu",
)
fig.show()

attr1_3    sinc1_3   intel1_3     fun1_3     amb1_3    shar1_3
gender                                                                  
Man     28.742112  15.837674  19.083084  16.666848   9.102337  11.007197
Woman   20.849198  17.197799  19.677666  15.881768  12.354964  14.071823

In [156]:
# Question 3/4 semaines : Que cherchez vous chez le sexe opposé? comparé à avant le Date, et au lendemain du Date
df_attr9 = df_unique[['attr1_2','sinc1_2','intel1_2','fun1_2','amb1_2','shar1_2','gender',
               'attr1_1','sinc1_1','intel1_1','fun1_1','amb1_1','shar1_1',
               'attr1_3','sinc1_3','intel1_3','fun1_3','amb1_3','shar1_3']]
moyenne_genre=df_attr9.groupby('gender').mean()
display(moyenne_genre)
df_attr9=df_attr9.drop('gender', axis=1)
moyennes_3_4 = df_attr9.mean().sort_values()

plot_data_list = []

# On réorganise les données pour le graphique
prefixes = ['attr', 'sinc', 'intel', 'fun', 'amb', 'shar']
moments = ['3/4 semaines après', 'Lendemain du Date', 'Avant le premier Date']  

for prefix in prefixes:
    for moment, suffix in zip(moments, ['1_3','1_2', '1_1']):
        col_name = f"{prefix}{suffix}"
        plot_data_list.append({
            'Attribute': prefix,
            'Moment': moment,
            'Value': moyennes_3_4[col_name]
        })

# Créer le DataFrame pour le graphique
plot_data = pd.DataFrame(plot_data_list)

fig = px.bar(
    plot_data,
    x='Attribute',
    y='Value',
    color='Moment',
    barmode='group',
    labels={'Attribute': 'Attribut', 'Value': 'Moyenne'},
    title='Comparaison sur 3 périodes de la répartition moyenne des points donnés aux attributs influençant la décision',
    color_discrete_sequence=['orange','blue', 'green']
)

fig.show()

attr1_2    sinc1_2   intel1_2     fun1_2     amb1_2    shar1_2  \
gender                                                                     
Man     30.126721  15.204534  16.751457  17.736356   8.535385  11.512024   
Woman   21.862996  16.576639  19.118109  17.489874  11.446849  13.942521   

          attr1_1   sinc1_1   intel1_1     fun1_1     amb1_1    shar1_1  \
gender                                                                    
Man     27.248800  16.37960  19.387418  17.600839   8.823956  10.982271   
Woman   18.020372  18.22223  20.971004  17.299108  12.818476  12.697836   

          attr1_3    sinc1_3   intel1_3     fun1_3     amb1_3    shar1_3  
gender                                                                    
Man     29.182308  15.980171  18.947265  16.386923   8.975983  10.870000  
Woman   20.565685  17.313767  19.846712  15.851301  12.294178  14.129589

#### 3/4 semaines après les résultats se réalignes plus sur les réponses du avant le Date. L'attirance reste bien sur le critère dominant, suivi de l'intelligence. Le "drôle/amusant" perd de son importance et les intérêts commun sont plutot stable suite au lendemain du date et donc confirme que ce point à pris de l'importance.

In [157]:
moyennes_3_4=pd.DataFrame(moyennes_3_4).T
moyennes_3_4

amb1_2     amb1_1     amb1_3    shar1_1    shar1_3    shar1_2    sinc1_2  \
0  9.964103  10.806476  10.818023  11.832126  12.679506  12.704722  15.877856   

      fun1_3    sinc1_3    sinc1_1     fun1_1     fun1_2   intel1_2  \
0  16.089582  16.720494  17.290754  17.451363  17.615402  17.912825   

    intel1_3   intel1_1    attr1_1    attr1_3    attr1_2  
0  19.446578  20.170478  22.685478  24.398935  26.080227

In [158]:
ecart_attra = (((moyennes_3_4.iloc[0, 16]*100/moyennes_3_4.iloc[0, 15])-100).round(2))
print(f"L'interet pour l'attirance évolue de {ecart_attra} % 3/4 semaines après les Dates")
ecart_attra_woman=(((moyenne_genre.iloc[1, 12]*100/moyenne_avant.iloc[1, 0])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_attra_woman} %")
ecart_attra_man=(((moyenne_genre.iloc[0, 12]*100/moyenne_avant.iloc[0, 0])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_attra_man} %")

print("")

ecart_sinc= (((moyennes_3_4.iloc[0, 8]*100/moyennes_3_4.iloc[0, 9])-100).round(2))
print(f"L'interet pour la sincérité évolue de {ecart_sinc} % 3/4 semaines après les Dates")
ecart_sinc_woman=(((moyenne_genre.iloc[1, 13]*100/moyenne_avant.iloc[1, 1])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_sinc_woman} %")
ecart_sinc_man=(((moyenne_genre.iloc[0, 13]*100/moyenne_avant.iloc[0, 1])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_sinc_man} %")

print("")

ecart_intel= (((moyennes_3_4.iloc[0, 13]*100/moyennes_3_4.iloc[0, 14])-100).round(2))
print(f"L'interet pour l'intelligence évolue de {ecart_intel} % 3/4 semaines après les Dates")
ecart_intel_woman=(((moyenne_genre.iloc[1, 14]*100/moyenne_avant.iloc[1, 2])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_intel_woman} %")
ecart_intel_man=(((moyenne_genre.iloc[0, 14]*100/moyenne_avant.iloc[0, 2])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_intel_man} %")

print("")

ecart_fun= (((moyennes_3_4.iloc[0, 7]*100/moyennes_3_4.iloc[0, 10])-100).round(2))
print(f"L'interet pour le 'drole/amusant' évolue de {ecart_fun} % 3/4 semaines après les Dates")
ecart_fun_woman=(((moyenne_genre.iloc[1, 15]*100/moyenne_avant.iloc[1, 3])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_fun_woman} %")
ecart_fun_man=(((moyenne_genre.iloc[0, 15]*100/moyenne_avant.iloc[0, 3])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_fun_man} %")

print("")

ecart_amb= (((moyennes_3_4.iloc[0, 2]*100/moyennes_3_4.iloc[0, 1])-100).round(2))
print(f"L'interet pour l'ambition évolue de {ecart_amb} % 3/4 semaines après les Dates")
ecart_amb_woman=(((moyenne_genre.iloc[1, 16]*100/moyenne_avant.iloc[1, 4])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_amb_woman} %")
ecart_amb_man=(((moyenne_genre.iloc[0, 16]*100/moyenne_avant.iloc[0, 4])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_amb_man} %")

print("")

ecart_shar= (((moyennes_3_4.iloc[0, 4]*100/moyennes_3_4.iloc[0, 3])-100).round(2))
print(f"L'interet pour l'intérêt commun évolue de {ecart_shar} % 3/4 semaines après les Dates")
ecart_shar_woman=(((moyenne_genre.iloc[1, 17]*100/moyenne_avant.iloc[1, 5])-100).round(2))
print(f"L'ecart chez les femmes est de {ecart_shar_woman} %")
ecart_shar_man=(((moyenne_genre.iloc[0, 17]*100/moyenne_avant.iloc[0, 5])-100).round(2))
print(f"L'ecart chez les hommes est de {ecart_shar_man} %")

L'interet pour l'attirance évolue de 7.55 % 3/4 semaines après les Dates
L'ecart chez les femmes est de 14.13 %
L'ecart chez les hommes est de 7.09 %

L'interet pour la sincérité évolue de -3.3 % 3/4 semaines après les Dates
L'ecart chez les femmes est de -4.97 %
L'ecart chez les hommes est de -2.44 %

L'interet pour l'intelligence évolue de -3.59 % 3/4 semaines après les Dates
L'ecart chez les femmes est de -5.36 %
L'ecart chez les hommes est de -2.28 %

L'interet pour le 'drole/amusant' évolue de -7.8 % 3/4 semaines après les Dates
L'ecart chez les femmes est de -8.37 %
L'ecart chez les hommes est de -6.89 %

L'interet pour l'ambition évolue de 0.11 % 3/4 semaines après les Dates
L'ecart chez les femmes est de -4.1 %
L'ecart chez les hommes est de 1.77 %

L'interet pour l'intérêt commun évolue de 7.16 % 3/4 semaines après les Dates
L'ecart chez les femmes est de 11.26 %
L'ecart chez les hommes est de -1.0 %


#### En comparant les attentes chez le sexe oppsosé entre avant le Date et 3/4 semaines après le Date, le critère sur l'attractivité est confirmé chez nos 2 partis avec une augmentation plus siginifactive chez les femmes. L'intelligence et le "drole/amusant" perdent le plus au profit notamment de l'intérêt en commun qui progresse de 7.21% nettement plébicité par les femmes, quand les hommes semblent plus sensible à l'ambition. 

## Poids des intérêts commun dans les Matchs
#### Comme nous avons pu le voir plus haut la race ne favorise pas les Matchs.
#### Nous venons de voir que les intérêts commun semblaient avoir plus d'importance dans la prise de décision et principalement dans le cadre d'un Non à une prochaine rencontre. Essayons de voir si ils ont plus de poids que la race dans les Matchs.
#### Nous allons utiliser la colonne int_corr qui donne la corrélation des intérêts commun de -1 à 1. 1 étant bien sur une parfaite corrélation des intérêts commun.   

In [159]:


# int_corr selon match
df_interet = df.copy()
df_interet['match'] = df['match'].replace({0: 'Pas de match', 1: 'Match'})
# configuration code couleur de la box
color_box= {"Match" : "green", "Pas de match" : "black"}
fig2 = px.box(df_interet, x='match', y='int_corr',
              title="Similarité des intérêts selon le match",
              labels={'int_corr': 'Corrélation des intérêts', 'match_label': 'Résultat des Matchs'},
              color='match', 
              color_discrete_map=color_box)
fig2.show()


#### Les intérêts communs ont un faible impact dans le succès des matchs. Néanmoins plus les goûts sont proches, plus les chances de match augmentent, mais comme vu plus haut ce n'est pas un critère déterminant à lui seul.


## Les gens peuvent-ils prédire avec précision leur propre valeur perçue sur le marché des rencontres ?
#### Nous allons créer une nouvelle colonne afin de calculer un indice de corrélation permettant de voir si les candidats ont une évaluation proche de l'évaluation du Date rencontré. Pour cela nous utiliserons les colonnes attr_o ... amb_o correspondant aux notes données par le Date et attr5_1 ... amb5_1 correspondant aux notes données soit même.


In [160]:
# on garde que les colonnes necessaires à notre calcule
cols = ['iid', 'attr_o', 'sinc_o', 'intel_o', 'fun_o', 'amb_o', 
        'attr5_1', 'sinc5_1', 'intel5_1', 'fun5_1', 'amb5_1']
# on supprime les NaN
df_corr = df[cols].dropna()

# On crée une fonction pour calculer la corrélation entre les attributs
def compute_corr(row):
    date = [row['attr_o'], row['sinc_o'], row['intel_o'], row['fun_o'], row['amb_o']]
    perso = [row['attr5_1'], row['sinc5_1'], row['intel5_1'], row['fun5_1'], row['amb5_1']]
    # l'utilisation dela fonction corr necessite d'utiliser des données en Series
    return pd.Series(date).corr(pd.Series(perso))

# Appliquer à chaque ligne
df_corr['alignement_corr'] = df_corr.apply(compute_corr, axis=1)

# Résumé statistique
print(df_corr['alignement_corr'].describe(include='all'))

/home/m_bar/miniconda3/envs/tinder/lib/python3.10/site-packages/numpy/lib/_function_base_impl.py:3045: RuntimeWarning:

invalid value encountered in divide

/home/m_bar/miniconda3/envs/tinder/lib/python3.10/site-packages/numpy/lib/_function_base_impl.py:3046: RuntimeWarning:

invalid value encountered in divide



count    3875.000000
mean        0.242811
std         0.493666
min        -1.000000
25%        -0.132187
50%         0.311805
75%         0.665133
max         1.000000
Name: alignement_corr, dtype: float64


In [164]:
# Sélectionner uniquement les colonnes utiles
cols = ['attr_o', 'sinc_o', 'intel_o', 'fun_o', 'amb_o', 
        'attr5_1', 'sinc5_1', 'intel5_1', 'fun5_1', 'amb5_1']

# Supprimer les NaN
df_corr = df[cols].dropna()


corr_matrix = df_corr.corr()

fig = px.imshow(
    corr_matrix,
    text_auto=True,         
    color_continuous_scale="RdBu_r", 
    zmin=-1, zmax=1,         
    title="Matrice de corrélation entre auto-évaluation et notes données"
)

fig.update_layout(
    width=800, height=600,
    xaxis_title="Variables",
    yaxis_title="Variables"
)

fig.show()

Les corrélations entre auto-évaluation et notes données sont proches de 0, donc très faibles. Cela signifie que les participants n’arrivent pas à bien se juger, ou que l’image qu’ils ont d’eux-mêmes ne correspond pas à la réalité perçue par les autres. On observe en revanche des corrélations entre certaines notes : lorsqu’une personne s’auto-évalue comme sincère, elle a tendance à se juger aussi plus intelligente. Côté évaluation des partenaires, on retrouve le même phénomène : une personne jugée attirante est souvent perçue aussi comme drôle ou intelligente

## Voyons maintenant si etre le premier ou dernier rendez vous de la soirée à un impact sur le Match

In [162]:
# Match selon l'ordre des rencontres
match_by_order = df.groupby("order")["match"].mean().reset_index()

fig = px.line(match_by_order, x="order", y="match", markers=True,
              title="Taux de match selon l'ordre des rencontres",
              labels={"match": "Taux de match", "order": "Ordre des rencontres"})

fig.show()

# Statistiques
correlation = df["order"].corr(df["match"])
first_five = df[df["order"] <= 5]["match"].mean()
last_five = df[df["order"] >= 17]["match"].mean()

print("Corrélation ordre/match:", correlation.round(2))

print("Match des 5 premières rencontres:", first_five.round(2))
print("Match des 5 dernières rencontres:", last_five.round(2))


Corrélation ordre/match: -0.04
Match des 5 premières rencontres: 0.18
Match des 5 dernières rencontres: 0.15


Première rencontre et taux de match le plus élevé de la soirée. Stress et manque de critère de référence ont du inciter des matchs prématurés. Au second rendez vous on retombe vite et on commence à mieux jauger les rencontres et etre plus en phase avec les critères. Au fur et à mesure que la soirée avance les matchs diminuent malgré des pics à la 9eme et 19eme places. Comportement étrange à la 21eme où l'on a le plus faible taux de match et la 22eme et dernière rencontre de la soirée où l'on remonte en flèche pour etre très proche de notre taux le plus élevé qui pour rappel est la première rencontre.
La corrélation entre l’ordre et le taux de match est de -0,04 : elle est très faible, mais suggère une légère tendance à avoir plus de matchs en début qu’en fin de soirée.
Regardons comment évolue la note des attributs au cours de la soirée pour savoir si certains de ces pics s'expliques. 

In [163]:
# Colonnes des notes
cols = ["attr_o", "sinc_o", "intel_o", "fun_o", "amb_o", "shar_o"]

# Évolution des notes selon l'ordre des rencontres
evolution = df.groupby("order")[cols].mean().reset_index()

fig = px.line(evolution, x="order", y=cols,
              title="Évolution des notes selon l'ordre des rencontres",
              labels={"value": "Note moyenne", "order": "Ordre des rencontres", "variable": "Les attributs"})

fig.show()

On peut constater que les notes des attributs reflètent bien la tendance des matchs, même si pour nos remarques à la 9eme et 19eme rencontre cela ne se voit pas forcement très bien. Nous pouvons le voir plus nettement sur la 15,16,17eme rencontre où la variation des notes des attributs collent bien à la variation du taux de match. Comme vu aussi plus haut la première note donnée aux attributs est très élevée, le manque de référence influe là aussi. Concernant la dernière note tous les critères chutent sauf le "drole/amusant" qui remonte en flèche. Le dernier Date a du être moins pertinent dans la décision du match car certainement que seul ce dernier critère a du faire pencher la balance.

# Pour conclure
Le dataset contient de nombreuses valeurs manquantes, ce qui a nécessité un traitement attentif.

Les critères d’évaluation diffèrent au début entre hommes et femmes, mais finissent par converger vers deux attributs centraux :

l’attirance,

l’intelligence.

L’attirance, peu prioritaire pour les femmes au départ (contrairement aux hommes), prend progressivement de l’importance au fil de la soirée.

Pour les hommes, c’est l’intelligence qui gagne en poids décisionnel.

L’appartenance à la même race n’est pas un facteur déterminant. En revanche, le partage d’intérêts communs améliore légèrement les chances de match.

Les candidats peinent globalement à s’auto-évaluer sur les attributs, même si certains y parviennent mieux que d’autres.

Être parmi les premiers rendez-vous offre une très légère probabilité supplémentaire d’obtenir un match. En fin de soirée, les taux de match deviennent plus instables, avec de fortes variations positives et négatives, possiblement liées à la fatigue ou à une baisse d’attention.
